policy/policy.sample.json

```diff
@@ -0,0 +1,70 @@
+{
+    "SecurityPolicies": [
+        {
+            "Name": "policy1", # names can be same for multiple tenants thus use uuid.
+            "PolicyID": "3710940D-FBC0-4A44-A95D-E549E628B4DF", # unique identifier across tenants.
+            "Associations": [ # can attach multiple tenants associations in case you want blanket policy support.
+                {
+                    "Tenant":"tenant1", #optional if TenantID below is provided 
+                    "TenantID": "3A6E467F-370B-43EA-BCEC-C28912EDD752", # optional but good to have to relate things as names are not unique
+                    "Segment": "Segment1", # optional
+                    "SegmentID": "01CA06A0-429B-4EE1-9185-F17D5CE4D0F2",
+                },
+                {
+                    "Tenant":"tenant1",
+                    "Segment": "Segment2",
+                },
+            ],
+            "Tags": [
+                {
+                    "Key": "Platform",
+                    "Value": "Openstack",
+                },
+                {
+                    "Key": "TenantID",
+                    "Value": "D46FD18B-A381-4297-A6D4-9DA594EE71C8",
+                },
+            ], # meta data attached to policies for various external enviorments like openstack/kybernetes
+            "Direction" : "Egress", # can be Egress or Ingress.
+            "CidrBlock": "0.0.0.0/0", # IP from L3 header
+            "Rules": [
+                {
+                    "Protocol": "ICMP",
+                    "FromPort": 4000,
+                    "ToPort": 4001,
+                    "IcmpTypeCode": [0, 8], # we look for this only if protocol above is 17, else ignore.
+                },
+                {
+                    "Protocol": "UDP",
+                    "FromPort": 8000,
+                    "ToPort": 8100,
+                    "IcmpTypeCode": [0, 8], # Ignore
+                },
+            ],
+            "IsStateful": true,
+            "Description": "hello there, security policies are fun!",
+        },
+        {
+            "Name": "policy2",
+            "PolicyID": "", # if none is given, generate one here.
```

Comment:
But the object-create-and-check cycle only needs to be written once and can then be re-used in different tests.

In the end you still need to test how the system behaves if the client did not specify the UUID. And then you need to test what happens when both clients specify the same UUID, etc. Of if the server for some reason didn't like the UUID provided by the client. All those tests you didn't have to write if the client could not provide the UUID.

And what if one day we change the format of the UUID. What if we make it "context aware" in the same way that we made our IP addresses context aware. We use a fancy storage system so that the UUID can tell us where the things is stored... Or we have certain prefixes to encode other bits of information that we need... except now we can't because every client can make up their own UUID. 

Just saying, weirder things have happened.